fixes #213 - created a slide in drawer for main page categories (mobile version)

djmgit opened the pull request:
fixes issue #213. Created a slide in drawer for categories
in start page for mobile version.
Indented and cleaned index.html

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

djmgit commented:
Test link : loklakapps2.surge.sh

![screenshot_2017-06-23-22-02-31-716_com android chrome](https://user-images.githubusercontent.com/16368427/27491690-024ae274-5861-11e7-9252-2bcb14a20150.png)
![screenshot_2017-06-23-22-02-56-710_com android chrome](https://user-images.githubusercontent.com/16368427/27491692-0252793a-5861-11e7-835c-86ffdacfe9d5.png)
![screenshot_2017-06-23-22-03-10-992_com android chrome](https://user-images.githubusercontent.com/16368427/27491691-02527192-5861-11e7-8a85-8a12c470f3b5.png)
![screenshot_2017-06-23-22-03-48-064_com android chrome](https://user-images.githubusercontent.com/16368427/27491693-025e60c4-5861-11e7-90e2-3abe3bf9c811.png)

djmgit commented:
@SKrPl @kavithaenair @Achint08 @hemantjadon open for review

hemantjadon commented:
@djmgit Test link?

singhpratyush commented:
@hemantjadon: http://loklakapps2.surge.sh

Achint08 commented on css/dashboard.css:
```diff
@@ -72,20 +56,38 @@ ul.request {
+  cursor: pointer;
+  transition: all 0.5s;
+  z-index: 2;
+  transition: all 0.3s;
```
Use just one transition

hemantjadon replied:
@djmgit don't use `all` in transitions, use comma separated transitions.

djmgit commented:
@hemantjadon @Achint08 made the changes

singhpratyush commented:
@djmgit: Please avoid using `png` where `glyphicon` can be used. Also, two hamburger menus look a bit weird here. What do you think of this?
![loklakapps2 surge sh- iphone 6 plus](https://user-images.githubusercontent.com/10860278/27651591-7d8e1dae-5c56-11e7-9b22-f348be82f4af.png)

* It is using `glyphicon` instead of image.
* It is not covering any app in the view.
* It is at the top of the side drawer instead of somewhere in between.

djmgit commented:
@hemantjadon thanks a lot for the suggestions, I will implement them ASAP

hemantjadon commented on css/dashboard.css:
```diff
@@ -72,20 +56,37 @@ ul.request {
+  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
+  cursor: pointer;
+  z-index: 2;
+  transition: top 0.3s;
```
Nope, only stick to transiotion of `transform` and `opacity` as they don't trigger `Paint`.

djmgit commented:
@singhpratyush @hemantjadon made the changes.

![sh](https://user-images.githubusercontent.com/16368427/27677901-eb2737ea-5cd0-11e7-971b-3b6ee6d687a3.png)

hemantjadon commented:
Okay one interesting thing I just noticed, is when you change the device width from small to large the drawer doesn't come up automatically, I wonder why it is so? Please change this behaviour.
Try resizing the window.

djmgit commented:
Test link : http://loklakapps2.surge.sh/

@singhpratyush I have fixed the menu button ( at the position where it is shown in the previous screen shot ). Also I have removed the overlapping problem however I had to use !important to do it. I followed this article here https://coderwall.com/p/wpjw4w/change-the-bootstrap-navbar-breakpoint . It is a kind of hacky workaround since bootstrap3 does not provide any easy way to change the breakpoint where hamburger appears.

@hemantjadon when the screen gets smaller I am using javascript to control the sidebar. On mobile screen its not a problem (as there is no option of enlarging the screen) but on browser, if some one squeezes the screen, uses the sidebar, closes it and then again expands the screen then the sidebar is gone as js is overriding css media query which it will obviously do. I am trying to find a solution for it. I can either remove the js part and use css animation or I can remove the media query part and use js window resize listener. Which one do you suggest. Also do let me know if there is a better way to do it.
Also should I create a separate issue for it. As this PR has already become quite long.

hemantjadon commented:
@djmgit Let's Iterate here not making this too Long.
Open up another issue. There are better approaches for sidenavs, I can code up some sample sidebar or some reference. Let's iterate here :) 👍
